core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -468,6 +486,20 @@ public Builder uri(String path) {
       return this;
     }
 
+    public Builder withProxy(String hostname, int port) {
+      Preconditions.checkNotNull(hostname, "Invalid hostname for http client proxy: null");
+      this.proxy = new HttpHost(hostname, port);
+      return this;
+    }
+
+    public Builder withProxyBasicAuthentication(String username, String password) {
```

Comment:
Would a better API just be passing the `CredentialsProvider`? That way it's more generic than just having a specific one for basic authentication. The user can construct their own and pass it in.

```
public Builder withProxyCredentialsProvider(CredentialsProvider);
```

Reply:
Fixed. Thanks for suggesting this.